packages/shell-api/src/shard.ts

```diff
@@ -424,4 +426,29 @@ export default class Shard extends ShellApiWithMongoClass {
     }
     return this.stopBalancer();
   }
+
+  @returnsPromise
+  @apiVersions([])
+  @serverVersions(['6.1.0', ServerVersions.latest])
+  @returnType('AggregationCursor')
+  async getShardedDataDistribution(): Promise<AggregationCursor> {
+    this._emitShardApiCall('getShardedDataDistribution', {});
+
+    const cursor = await this._database.aggregate([{ $shardedDataDistribution: {} }]);
```

Comment:
I assume it doesn’t matter which specific database the aggregation is being run against?

Reply:
With new approach, it must be run in the admin database!